conda/testing/solver_helpers.py

```diff
@@ -0,0 +1,1280 @@
+from __future__ import annotations
```

Comment:
Not sure if the overlap of `conda/testing/solver_helpers.py` with `conda/testing/helpers.py` is useful, maybe worth doing a sub-package like `conda.testing.helpers.solvers` or simply `conda.testing.solvers`?

Reply:
I was hesitating whether to merge or not, but given the history of that file (and how radically different the approaches are; @FFY00 did an AWESOME work with that test interface), I decided against that. 

Let me know how you want to proceed here:

* Leave files separate but rename/move to make it more clear
* Perform surgery on `conda.testing.helpers` and move things that concern solvers to `conda.testing.solver`

Reply:
Can we tackle this in a different PR?

Reply:
Sounds good!